modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -484,22 +484,22 @@ private CompletableFuture<?> processOperationRequest(ReplicaRequest request, @Nu
      */
     private CompletableFuture<LeaderOrTxState> processTxStateCommitPartitionRequest(TxStateCommitPartitionRequest request) {
         return placementDriver.getPrimaryReplica(replicationGroupId, hybridClock.now())
-                .thenCompose(primaryReplica -> {
-                    if (primaryReplica == null) {
-                        return failedFuture(new PrimaryReplicaMissException(localNode.name(), null));
+                .thenCompose(primaryReplicaMeta -> {
```

Comment:
Why?

Reply:
Because the future does not return the primary replica, but the meta of the primary replica (`org.apache.ignite.internal.placementdriver.ReplicaMeta`).
Without the "meta" postfix, the variable name confuses me.

Reply:
argh, ok